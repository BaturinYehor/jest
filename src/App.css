.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.wrapper {
height: 100vh;
display: flex;
justify-content: center;
align-items: center;

background: rgba(235,23,71,1);
background: -moz-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -webkit-gradient(left top, left bottom, color-stop(0%, rgba(235,23,71,1)), color-stop(59%, rgba(235,23,71,1)), color-stop(60%, rgba(0,172,238,1)), color-stop(60%, rgba(0,172,238,1)), color-stop(100%, rgba(0,172,238,1)));
background: -webkit-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -o-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: -ms-linear-gradient(top, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
background: linear-gradient(to bottom, rgba(235,23,71,1) 0%, rgba(235,23,71,1) 59%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 60%, rgba(0,172,238,1) 100%);
filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#eb1747', endColorstr='#00acee', GradientType=0 );

}

.main {
  height: 60vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.main {
  height: 80vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.greeting {
  height: 60vh;
  background: #e6e6e6;
  box-shadow: 0 0 50px -10px rgba(0,0,0,1.5);
  width: 60%;
  padding: 30px;
  border-radius: 10px;
}

.text {
  color: #424242;
  font-family: "Adobe Caslon Pro", "Hoefler Text", Georgia, Garamond, Times, serif;
  letter-spacing:0.1em;
  text-align:center;
  text-transform: lowercase;
  font-size: 14pt;
  font-variant: small-caps;
}

.form {
  height: 80vh;
  padding-top: 100px;
  padding-left: 50px;
  border-top: 2px solid #00ACEE;
  border-bottom: 2px solid #00ACEE;
  border-right: 2px solid #00ACEE;
}

input[type="text"] {
  color: #424242;
  background-color: transparent;
  border: 0;
  border-bottom: solid 2px #EB1747;
  width: 50%;
  padding-bottom: 4px;
  font-weight: lighter;
  margin-bottom: 30px;
  margin-right: 20px;
  font-size: 20px;
  outline: none;
}

.get-name-button {
  position: absolute;
  left:    0;
  bottom:   0;
}

.get-name-form {
  position:relative;
  margin-top: 50px;
  height: 45vh;
}

.img {
  height: 15vh;
  width: 15vh;
  background-blend-mode: multiply;
  background: #e6e6e6 url("nix.png") no-repeat center center;
  background-size: contain;
  position: absolute;
  left:    0;
  bottom:   0;
}

input::-webkit-input-placeholder {
  color: #00ACEE;
}

input::-moz-placeholder {
  text-transform: lowercase;
}

button {
  border: 2px solid #EB1747;
  border-bottom: 4px solid #EB1747;
  padding: 8px 20px;
  border-radius: 3px;
  cursor: pointer;
  background-color: #EB1747;
  color: #fff;
  outline: none!important;
}

button:hover {
  border-bottom: 2px solid #00ACEE;
  position: relative;
  top: 2px;
}

.error {
  color: #EB1747;
  font-weight: bold;
}
