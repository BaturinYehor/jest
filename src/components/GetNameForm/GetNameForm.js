import React from "react";

class GetNameForm extends React.Component {

    constructor(props) {
        super(props);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.props.handleSubmit}>
                    <div className="get-name-form">
                        <div className="row">
                            <input type="text" name="name" placeholder="Name"/>
                        </div>
                        <div className="row h-75">
                            <div className="col-sm-10">
                                <div className="get-name-button">
                                    <button>Get information</button>
                                </div>
                            </div>
                            <div className="col-sm-2">
                                <div className="img"/>
                            </div>
                        </div>
                    </div>
                </form>
            </div>
        );
    }
}

export default GetNameForm