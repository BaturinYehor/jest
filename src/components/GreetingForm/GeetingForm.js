import React from "react";
import GetNameForm from "../GetNameForm/GetNameForm"
import {Redirect} from "react-router";

class GreetingForm extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            error: ""
        }
    };

    handleSubmit = (e) => {
        e.preventDefault();

        let name = e.target['name'].value;
        if (name) {
            this.props.history.setState({name: name});
        } else {
            this.setState({error: "Name can't be empty"})
        }
    };

    render() {
        if (this.props.history && this.props.history.state.name) {
            return <Redirect to={{pathname: '/main'}}/>
        }

        return (
            <div>
                <div className="nix-background">
                    <div className="card-form">
                        <div className="text">
                            Give us your name and let's get straight to it!
                        </div>
                        <GetNameForm handleSubmit={this.handleSubmit}/>
                        <div className="error">
                            {this.state.error}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default GreetingForm