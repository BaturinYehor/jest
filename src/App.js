import React from 'react';
import {Route} from "react-router-dom"
import {BrowserRouter} from 'react-router-dom/cjs/react-router-dom.min';
import GreetingForm from './components/GreetingForm/GeetingForm';
import Main from './components/Main/Main';

class
App extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            name: ''
        }
    };

    render() {
        return (
            <BrowserRouter>
                <Route
                    path='/greeting'
                    component={() => <GreetingForm history={this}/>}
                />
                <Route
                    path='/main'
                    component={() => <Main name={this.state.name}/>}
                />
            </BrowserRouter>
        )
    }
}

export default App