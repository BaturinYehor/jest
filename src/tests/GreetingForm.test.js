import React from 'react';
import renderer from 'react-test-renderer';
import GetNameForm from '../components/GetNameForm/GetNameForm'
import GreetingForm from '../components/GreetingForm/GeetingForm'
import App from '../App'
import {mount} from 'enzyme';
import {MemoryRouter} from 'react-router-dom/cjs/react-router-dom.min';

describe('GreetingForm test suite', () => {

    it('GreetingForm should match snapshot', () => {
        const component = renderer.create(<GreetingForm/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('GreetingForm form should have exactly one GetNameForm', () => {
        const wrapper = mount(<GreetingForm/>);
        expect(wrapper.find(GetNameForm).length).toEqual(1);
    });

    it('GreetingForm text should be correct', () => {
        const wrapper = mount(<GreetingForm/>);
        expect(wrapper.find(".text").text()).toEqual('Give us your name and let\'s get straight to it!');
    });

    it('Error should arise when passing empty name', () => {
        let expectedName = "Name can't be empty";
        const wrapper = mount(<GreetingForm/>);
        const e = {
            preventDefault: () => {
            },
            target: {
                name: {
                    value: ""
                }
            }
        };
        wrapper.instance().handleSubmit(e);
        expect(wrapper.instance().state.error).toEqual(expectedName);
    });

    it('Name should be set', () => {
        const expectedName = "Nick";

        const appContainer = mount(<App/>);
        const greetingContainer = mount(<GreetingForm history={appContainer.instance()}/>);
        const e = {
            preventDefault: () => {
            },

            target: {
                name: {
                    value: expectedName
                }
            }
        };
        greetingContainer.instance().handleSubmit(e);
        expect(appContainer.instance().state.name).toEqual(expectedName);
    });

    it('should redirect to main', () => {
        const history = {
            state: {
                name: "somename"
            }
        };

        const wrapper = mount(
            <MemoryRouter>
                <GreetingForm history={history}/>
            </MemoryRouter>
        );
        const app = wrapper.find(GreetingForm).instance();
        expect(app.render().props.to.pathname).toBe("/main")
    });
});