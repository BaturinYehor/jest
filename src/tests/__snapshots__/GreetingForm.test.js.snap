// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite GreetingForm should match snapshot 1`] = `
<div>
  <div
    className="wrapper"
  >
    <div
      className="greeting"
    >
      <div
        className="text"
      >
        Give us your name and let's get straight to it!
      </div>
      <div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="get-name-form"
          >
            <div
              className="row"
            >
              <input
                name="name"
                placeholder="Name"
                type="text"
              />
            </div>
            <div
              className="row h-50"
            >
              <div
                className="col-sm-10"
              >
                <div
                  className="get-name-button"
                >
                  <button>
                    Get information
                  </button>
                </div>
              </div>
              <div
                className="col-sm-2"
              >
                <div
                  className="img"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        className="error"
      >
        
      </div>
    </div>
  </div>
</div>
`;
